src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -178,16 +204,23 @@ xccl_mhba_alltoall_init(xccl_coll_op_args_t *coll_args,
                         xccl_mhba_coll_req_t *request,
                         xccl_mhba_team_t *team)
 {
+    if (coll_args->buffer_info.len > 128){
+        xccl_mhba_error("msg size too long");
+        return XCCL_ERR_NO_RESOURCE;
+    }
     int asr_rank = 0; // TODO select?
     int is_asr = (team->node.sbgp->group_rank == asr_rank);
     int n_tasks = (!is_asr) ? 2 : 6;
     int i;
     xccl_schedule_init(&request->schedule, team->super.ctx);
     request->asr_rank = asr_rank;
+    request->block_size = team->blocks_sizes[__ucs_ilog2_u32(coll_args->buffer_info.len-1)];
     assert(asr_rank < team->node.sbgp->group_size);
     request->tasks = (xccl_mhba_task_t*)malloc(sizeof(xccl_mhba_task_t)*n_tasks);
     request->seq_num = team->sequence_number;
     team->sequence_number++;
+    while(team->occupied_operations_slots[seq_index(team->sequence_number)]){} //wait for slot to be open
```

Comment:
blocking while loop - will cause a dead lock

Reply:
move to fanin start interface